astropy/utils/masked/function_helpers.py

```diff
@@ -133,10 +133,18 @@
 }  # fmt: skip
 
 # Really should do these...
-IGNORED_FUNCTIONS |= {
-    getattr(np, setopsname) for setopsname in np.lib.arraysetops.__all__
-}
-
+if NUMPY_LT_2_0:
+    NP_LIB_ARRAYSETOP_FUNCS = np.lib.arraysetops.__all__
+else:
+    # Public set operations have been moved to the top-level namespace in numpy 2.0
+    # (numpy/numpy#24507), raising an AttributeError when accessed through np.lib.arraysetops.
+    # Listing them explicitly here.
+    NP_LIB_ARRAYSETOP_FUNCS = [
+        'ediff1d', 'intersect1d', 'setxor1d', 'union1d', 'setdiff1d', 'unique',
+        'in1d', 'isin'
+    ]  # fmt: skip
+IGNORED_FUNCTIONS |= {getattr(np, setopsname) for setopsname in NP_LIB_ARRAYSETOP_FUNCS}
+del NP_LIB_ARRAYSETOP_FUNCS
```

Comment:
```suggestion
if NUMPY_LT_2_0:
    from np.lib import arraysetops
else:
    # Public set operations have been moved to the top-level namespace in numpy 2.0
    # (numpy/numpy#24507), raising an AttributeError when accessed through np.lib.arraysetops.
    from np.lib import _arraysetops_impl as arraysetops

IGNORED_FUNCTIONS |= {getattr(np, setopsname) for setopsname in arraysetops.__all__}
```

Reply:
@dhomeier I can't take your patch directly from the suggestion because `from np.lib ...` isn't valid (it should be `from numpy.lib ...`), but I'd be happy to push the correct version of it !
Other than that, it seems contradictory with [your other comment](https://github.com/astropy/astropy/pull/15253#discussion_r1310922130). Again, I'm happy to follow whatever consensus is reached.